earth_enterprise/src/common/FileAccessor.cpp

```diff
@@ -0,0 +1,88 @@
+#include "FileAccessor.h"
+
+std::unique_ptr<AbstractFileAccessor> AbstractFileAccessor::getAccessor(const std::string &fname) {
+  return std::unique_ptr<POSIXFileAccessor>(new POSIXFileAccessor());
+}
+
+// ****************************************************************************
+// ***  POSIXFileAccessor
+// ****************************************************************************
+int POSIXFileAccessor::Open(const std::string &fname, const char *mode, int flags, mode_t createMask) {
+  int result;
+  if (mode) {
+    if (strcmp(mode, "w") == 0) {
+      result = khOpenForWrite(fname, createMask);
+    }
+    else {
+      result = khOpenForRead(fname);
+    }
+  }
+  else {
+    result = khOpen(fname, flags, createMask);
+  }
+  fileDescriptor = result;
+  return result;
```

Comment:
Minor: do we need to return the file descriptor? It would be nice to keep it encapsulated in this class as much as possible.